src/commands/CmdCalendar.cpp

```diff
@@ -603,40 +607,53 @@ std::string CmdCalendar::renderMonths (
         if (today.sameDay (date))
           cellColor.blend (color_today);
 
-        // colorize due tasks
+        // colorize due and scheduled tasks
         if (config.get ("calendar.details") != "none")
         {
           config.set ("due", 0);
+          config.set ("scheduled", 0);
+          bool dueTaskFound = false;
```

Comment:
Is the purpose of this variable to prevent the "scheduled" and "overdue" colors being applied on the same date?

Reply:
yes, if the date has a task has a due date, tasks appearing later in the task array should not overwrite the due color, since I think due dates are more important to be marked in the calendar than scheduled ones

Reply:
Understood and agreed. Two suggestions/observations:
* for a decision like that (due takes precendence over scheduled when coloring), we should at least add a comment to ensure future readers of this code understand this was a deliberate decision made here :slightly_smiling_face: 
* I don't think the code works correctly, as is. The `all` vector contains all the tasks, not just the tasks that are scheduled/due on the given date. This means that while the code is evaluating the color for the, say, July 27th, it can encounter a task with due date of Dec 31st, and it will still flip the value of the `dueTaskFound` to `true`. This will in turn prevent that date from being colored with the scheduled coloring, if there is a matching task.

The flag should be flipped only after the color was definitely applied (that is, after the `cellColor.blend` calls). Also I'd suggest a slightly more descriptive variable name, perhaps `coloredWithDue`?

Reply:
Thanks, that is (maybe) correct. I still do not completely understand the code, I think:

```
              if(task.has("due")) {
                std::string due = task.get ("due");
                Datetime duedmy (strtol (due.c_str(), nullptr, 10));

                if (duedmy.sameDay (date))
                {
                  switch (task.getDateState ("due"))
                  {
                  case Task::dateNotDue:
                    break;
                    ...
```

If `duedmy.sameDay (date)` evaluates to true, how could the case Task::dateNotDue happen? I thought this should not happen and thus It shouuld have been fine to set the value of the coloredWithDue before the whole switch section, right?

Reply:
You are correct that the case `Task::dateNotDue` will never happen, I believe that is there for completeness (or perhaps compiler nagging).

The previous version of the code, however, did not enforce that the flag was flipped only if the task is due on the same day, it was flipped if the task had a due attribute defined (with any value):

![image](https://user-images.githubusercontent.com/2123868/126243662-bb549615-7126-45d4-8ef7-855b5d523d8c.png)

That was incorrect.

However, you're right that we could move the `coloredWithDue = true;` statement in front of the switch statement, which would simplify the code a bit. Good suggestion, I missed that :slightly_smiling_face:

Reply:
@sebu06 up to you if you want to refactor, I'm also happy merging as is (after the Changelog conflict is resolved).